electron-app/src/components/settings/api-keys/ExchangeSettings.vue

```diff
@@ -99,16 +105,22 @@ import BaseExternalLink from '@/components/base/BaseExternalLink.vue';
 import ConfirmDialog from '@/components/dialogs/ConfirmDialog.vue';
 import MessageDialog from '@/components/dialogs/MessageDialog.vue';
 import ExchangeBadge from '@/components/ExchangeBadge.vue';
+import { convertToGeneralSettings } from '@/data/converters';
 import { exchanges } from '@/data/defaults';
 import { Message } from '@/store/store';
+import { GeneralSettings } from '@/typing/types';
 
-const { mapState } = createNamespacedHelpers('balances');
+const { mapState } = createNamespacedHelpers('session');
 
 @Component({
   components: { ConfirmDialog, MessageDialog, ExchangeBadge, BaseExternalLink },
-  computed: mapState(['connectedExchanges'])
+  computed: {
+    ...mapState(['connectedExchanges']),
+    ...mapState(['settings'])
+  }
 })
 export default class ExchangeSettings extends Vue {
+  settings!: GeneralSettings;
```

Comment:
Instead of trying to map the whole settings object in the component you should add a getter instead at https://github.com/rotki/rotki/blob/08562c304c528f2d2bf2f8a7556a33d0dac0194d/electron-app/src/store/session/getters.ts#L24

```typescript
  krakenAccountType: (state: SessionState) => {
    return state.settings.krakenAccountType;
  },
```

Reply:
done